src/web.rs

```diff
@@ -655,6 +709,35 @@ impl ServiceInner {
         }
         Ok(None)
     }
+    fn save_sample_file_dir(&self, req: &Request<::hyper::Body>, body: serde_json::Value) -> ResponseResult {
+        let (mut resp, writer) = http_serve::streaming_body(&req).build();
+        resp.headers_mut().insert(header::CONTENT_TYPE,
+                                  HeaderValue::from_static("application/json"));
+        let mut db = self.db.lock();
+        match body["path"].is_string() {
```

Comment:
I think it'll be more readable to add a new type to `json.rs` and avoid all this `is_string()` + `unwrap()` stuff.